packages/integration-tests/projects/suite-web/tests/settings/custom-firmware.test.ts

```diff
@@ -0,0 +1,49 @@
+// @group:settings
+// @retry=2
+describe('Install custom firmware', () => {
+    beforeEach(() => {
+        cy.task('startEmu', { wipe: true });
+        cy.task('setupEmu');
+        cy.task('startBridge');
+        cy.viewport(1080, 1440).resetDb();
+        cy.prefixedVisit('/settings/device');
+        cy.passThroughInitialRun();
+    });
+    /*
+    *1. Navigate to `Settings/Device`
+    *2. Click on `Install firmware`
+    *3. Select the custom firmware
+        a)Observe the initialization on the device
+    *4. Complete the FW instalation on the device
+    */
+
+    it('go to device settings and check if custom FW modal appears', () => {
+        //
+        // Test preparation
+        //
+        cy.getTestElement('@settings/device/custom-firmware-modal-button').click({
+            force: true,
+        });
+        //
+        // Test execution
+        //
+        cy.get('[data-test="@firmware-custom"]').then(fileUploadModal => {
```

Comment:
using 
```
cy.getTestElement("@firmware-custom").then(...)
```
does not work here? maybe it indeed does not. If so, we should fix it. cy.getTestElement should have the same function signature as cy.get